quickstarts/microsoft.network/application-gateway-v2-autoscale-create/main.bicep

```diff
@@ -0,0 +1,172 @@
+
+@description('Maximum instance count for Application Gateway')
+param maxCapacity int = 10
+
+@description('Application Gateway Frontend port')
+param frontendPort int = 80
+
+@description('Application gateway Backend port')
+param backendPort int = 80
+
+@description('Back end pool ip addresses')
+param backendIPAddresses array = [
+  {
+    IpAddress: '10.0.0.4'
+  }
+  {
+    IpAddress: '10.0.0.5'
+  }
+]
+
+@allowed([
+  'Enabled'
+  'Disabled'
+])
+@description('Cookie based affinity')
+param cookieBasedAffinity string = 'Disabled'
+
+@description('Location for all resources.')
+param location string = resourceGroup().location
+
+var appGwPublicIpName = '${applicationGatewayName}-pip'
+var appGwSize = 'Standard_v2'
+
+resource virtualNetwork 'Microsoft.Network/virtualNetworks@2020-06-01' = {
+  name: virtualNetworkName
+  location: location
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        vnetAddressPrefix
+      ]
+    }
+    subnets: [
+      {
+        name: subnetName
+        properties: {
+          addressPrefix: subnetPrefix
+        }
+      }
+    ]
+  }
+}
+
+resource publicIP 'Microsoft.Network/publicIPAddresses@2020-06-01' = {
+  name: appGwPublicIpName
+  location: location
+  sku: {
+    name: 'Standard'
+  }
+  properties: {
+    publicIPAllocationMethod: 'Static'
+  }
+}
+
+resource applicationGateway 'Microsoft.Network/applicationGateways@2020-06-01' = {
+  name: applicationGatewayName
+  location: location
+  properties: {
+    sku: {
+      name: appGwSize
+      tier: 'Standard_v2'
+    }
+    autoscaleConfiguration: {
+      minCapacity: minCapacity
+      maxCapacity: maxCapacity
+    }
+    gatewayIPConfigurations: [
+      {
+        name: 'appGatewayIpConfig'
+        properties: {
+          subnet: {
+            id: resourceId('Microsoft.Network/virtualNetworks/subnets', virtualNetwork.name, subnetName)
+          }
+        }
+      }
+    ]
+    frontendIPConfigurations: [
+      {
+        name: 'appGatewayFrontendIP'
+        properties: {
+          publicIPAddress: {
+            id: publicIP.id
+          }
+        }
+      }
+    ]
+    frontendPorts: [
+      {
+        name: 'appGatewayFrontendPort'
+        properties: {
+          port: frontendPort
+        }
+      }
+    ]
+    backendAddressPools: [
+      {
+        name: 'appGatewayBackendPool'
+        properties: {
+          backendAddresses: backendIPAddresses
+        }
+      }
+    ]
+    backendHttpSettingsCollection: [
+      {
+        name: 'appGatewayBackendHttpSettings'
+        properties: {
+          port: backendPort
+          protocol: 'Http'
+          cookieBasedAffinity: cookieBasedAffinity
+        }
+      }
+    ]
+    httpListeners: [
+      {
+        name: 'appGatewayHttpListener'
+        properties: {
+          frontendIPConfiguration: {
+            id: resourceId('Microsoft.Network/applicationGateways/frontendIPConfigurations', applicationGatewayName, 'appGatewayFrontendIP')
```

Comment:
does the bicep symbolic reference work here?  (I'm guessing it fails with circular reference? if so we should probably file an issue since thie RP seems to require it)

Reply:
you're saying i should convert to .id?  Not sure how to do that here.

Reply:
I don't think we can here - I'm pretty sure bicep won't allow .id on this resource type...